nativelib/src/main/scala/scala/scalanative/runtime/Intrinsics.scala

```diff
@@ -29,4 +29,97 @@ object Intrinsics {
+
+  /** Intrinsified raw memory load of float. */
+  def loadFloat(rawptr: RawPtr): Float = intrinsic
+
+  /** Intrinsified raw memory load of double. */
+  def loadDouble(rawptr: RawPtr): Double = intrinsic
+
+  /** Intrinsified raw memory load of rawptr. */
+  def loadRawPtr(rawptr: RawPtr): RawPtr = intrinsic
+
+  /** Intrinsified raw memory load of object. */
+  def loadObject(rawptr: RawPtr): Object = intrinsic
+
+  /** Intrinsified raw memory store of boolean. */
+  def storeBoolean(rawptr: RawPtr, value: Boolean): Unit = intrinsic
+
+  /** Intrinsified raw memory store of char. */
+  def storeChar(rawptr: RawPtr, value: Char): Unit = intrinsic
+
+  /** Intrinsified raw memory store of byte. */
+  def storeByte(rawptr: RawPtr, value: Byte): Unit = intrinsic
+
+  /** Intrinsified raw memory store of short. */
+  def storeShort(rawptr: RawPtr, value: Short): Unit = intrinsic
+
+  /** Intrinsified raw memory store of int. */
+  def storeInt(rawptr: RawPtr, value: Int): Unit = intrinsic
+
+  /** Intrinsified raw memory store of long. */
+  def storeLong(rawptr: RawPtr, value: Long): Unit = intrinsic
+
+  /** Intrinsified raw memory store of float. */
+  def storeFloat(rawptr: RawPtr, value: Float): Unit = intrinsic
+
+  /** Intrinsified raw memory store of double. */
+  def storeDouble(rawptr: RawPtr, value: Double): Unit = intrinsic
+
+  /** Intrinsified raw memory store of rawptr. */
+  def storeRawPtr(rawptr: RawPtr, value: RawPtr): Unit = intrinsic
+
+  /** Intrinsified raw memory store of object. */
+  def storeObject(rawptr: RawPtr, value: Object): Unit = intrinsic
+
+  /** Intrinsified computation of derived raw pointer. */
+  def elemRawPtr(rawptr: RawPtr, offset: Long): RawPtr = intrinsic
+
+  /** Intrinsified cast that reinterprets raw pointer as an object. */
+  def castRawPtrToObject(rawptr: RawPtr): Object = intrinsic
+
+  /** Intrinsified cast that reinterprets object as a raw pointers. */
+  def castObjectToRawPtr(obj: Object): RawPtr = intrinsic
+
+  /** Intrinsified cast that reinterprets int as a float. */
+  def castIntToFloat(int: Int): Float = intrinsic
+
+  /** Intrinsified cast that reinterprets float as an int. */
+  def castFloatToInt(float: Float): Int = intrinsic
+
+  /** Intrinsified cast that reinterprets long as a double. */
+  def castLongToDouble(long: Long): Double = intrinsic
+
+  /** Intrinsified cast that reinterprets double as a long. */
+  def castDoubleToLong(double: Double): Long = intrinsic
+
+  /** Intrinsified cast that reinterprets raw pointer as an int. */
+  def castRawPtrToInt(rawptr: RawPtr): Int = intrinsic
+
+  /** Intrinsified cast that reinterprets raw pointer as an long. */
+  def castRawPtrToLong(rawptr: RawPtr): Long = intrinsic
+
+  /** Intrinsified cast that reinterprets int as a raw pointer. */
+  def castIntToRawPtr(int: Int): RawPtr = intrinsic
+
+  /** Intrinsified cast that reinterprets long as a raw pointer. */
+  def castLongToRawPtr(int: Long): RawPtr = intrinsic
```

Comment:
Curious reader might observe that this API maps very closely to memory operations in [`sun.misc.Unsafe`](http://www.docjar.com/docs/api/sun/misc/Unsafe.html).